cli/client/authentication.go

```diff
@@ -74,12 +76,43 @@ func (client *RestClient) Logout(token string) error {
 }
 
 // RefreshAccessToken returns a new access token given valid refresh token
-func (client *RestClient) RefreshAccessToken(token string) (*types.Tokens, error) {
-	res, err := client.R().
-		SetHeader("Content-Type", "application/json").
-		SetBody(map[string]string{"refresh_token": token}).
-		SetResult(&types.Tokens{}).
-		Post("/auth/token")
+func (client *RestClient) RefreshAccessToken(tokens *corev2.Tokens) (*corev2.Tokens, error) {
+	var err error
+	var res *resty.Response
+	url := "/auth/token"
+
+	// Parse the access claims so we can determine the issuer URL
+	parser := new(jwt.Parser)
+	claims := &corev2.Claims{}
+	_, _, _ = parser.ParseUnverified(tokens.Access, claims)
+
+	switch issuer := claims.Issuer; {
+	case issuer != "":
+		// Attempt to renew the access token with the issuer
+		res, err = client.R().
+			SetHeader("Content-Type", "application/json").
+			SetBody(map[string]string{"refresh_token": tokens.Refresh}).
+			SetResult(&corev2.Tokens{}).
+			Post(claims.Issuer + url)
```

Comment:
I understand why `iss` is a URL instead of just a host now. I'd like to suggest an alternative approach.

Default to `https`. If that fails, try on `http` and issue a big fat warning stating that this is completely insecure when using JWTs. That way `iss` is just the host and we always try to use `https` whenever we can. JWTs over http make as much security sense as passwords over http anyway.

Reply:
We can already detect whether the token was issued over HTTP or HTTPS. Right now we can't serve both HTTP _and_ HTTPS traffic over the same API port so trying to request the HTTPS version of an HTTP API that we already know is not secure doesn't do anything is most scenarios.

That being said, your comment about authentication over HTTP totally makes sense but I would argue that this PR doesn't change the current reality and to avoid a scope creep, I'd prefer to open a new issue so we can discuss about the right approach to take to encourage users and operators to use HTTPS.